exercises/armstrong-numbers/.meta/generator/armstrong_numbers_case.rb

```diff
@@ -0,0 +1,36 @@
+require 'generator/exercise_case'
+
+class ArmstrongNumbersCase < Generator::ExerciseCase
+
+  def workload
+    case expected
+    when true  then true_workload
+    when false then false_workload
+    else raise "unexpected 'expect' value encountered"
+    end
+  end
+
+  def true_workload
+    [ "assert", "#{call_armstrong}" ].join(' ')
+  end
+
+  def false_workload
+    [ "refute", "#{call_armstrong}" ].join(' ')
+  end
+
+  def call_armstrong
+    delimited_input = delimit_integers(input['number'])
+    "ArmstrongNumbers.include?(#{delimited_input})"
+  end
+
+  private
+
+  def delimit_integers(integer)
+    integer
+      .digits
+      .each_slice(3)
+      .map(&:join)
+      .join('_')
+      .reverse
```

Comment:
I guess it would actually be more efficient to `reverse` before `join`, because this way fewer things are reversed.  I think it would be `O(n / 3 + 1)`.

Does that matter you all?

Reply:
Nevermind, I just tried, but I forgot the order of the digits was reversed by digits (which is necessary), so it just messed things up :smile: Leaving this as it is.